### ISSUE TITLE
[BUG] Alert if agent is in vacation or blocked

### ISSUE THREAD

**Issue by antiuser** [osTicket/osTicket#5611]
### Prerequisites

* [NO] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [NO] Do you have any errors in the PHP error log, or javascript console?
* [YES] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [YES] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/osTicket/osTicket/wiki/Github-Issues-Guidelines)

### Description
When i block agent or mark agent as vaccation - then this osTicket send any notification (such as create a new ticket, reply to a ticket). I think if agent is blocked or vacation attribute present then notification to this agent should be disabled.  
### Steps to Reproduce

1. Install block/vacation attribute present to agent
2. Create any ticket to team (this agent stay in this team)
3. Agent in vacation or blocked agent receive notification on email with this new ticket

So testing agent is manager of this team

**Expected behavior:**
Im assume, if agent in vacation or blocked this agent shouldnt accept any notifications

**Actual behavior:** [What actually happened]
If agent in vacation or blocked - this agent accept any notifications on email
### Versions
osTicket 1.12.6
Admin panel -> Dashboard -> Information which also additionally gives you information about your server.

Also, please include the OS and what version of the OS you're running. As well as your browser and browser version.
Debian 9.x x64

![Снимок экрана_2020-08-13_19-39-05](https://user-images.githubusercontent.com/5499424/90148695-0c0cf080-dd9d-11ea-81c2-ed6b103550bd.png)
![Снимок экрана_2020-08-13_19-29-07](https://user-images.githubusercontent.com/5499424/90148100-66598180-dd9c-11ea-8dae-fd68c32d1334.png)
![Снимок экрана_2020-08-13_19-28-22](https://user-images.githubusercontent.com/5499424/90148144-71141680-dd9c-11ea-8905-d0c84723141c.png)
![Снимок экрана_2020-08-13_19-29-58](https://user-images.githubusercontent.com/5499424/90148195-825d2300-dd9c-11ea-90f9-5c30d60ceae3.png)

**Comment by JediKev** [comment]
@antiuser 

Please edit your post and fill out the issue template. You are leaving out key information that we need to assist. Once you fill out the issue template we will be able to better assist.

Also, if you could provide screenshots and step-by-step instructions on how to replicate that would be great as I have no idea what you mean from your OP.

Cheers.

**Comment by antiuser** [comment]
@JediKev edited, thank you :)

**Comment by JediKev** [comment]
@antiuser 

Try applying these changes:
- https://github.com/osTicket/osTicket/pull/5573

Cheers.

**Comment by JediKev** [comment]
@antiuser 

I have also checked everywhere we do the team member alerts and we loop through each staff member to ensure they are available (ie. not locked or on vacation) before sending the alert. 🤔 

If all else fails, I'd suggest upgrading to v1.14.2 and retest.

Cheers.

**Comment by ntozier** [comment]
@antiuser as a side note 1.12 is pretty old at this point.  You might want to just upgrade to the latest stable (1.14.2).

**Comment by antiuser** [comment]
@JediKev @ntozier im already apply changes, but can confirm this fix tomorrow only. Because our department going to rest today :) Thank you so much!

**Comment by antiuser** [comment]
@JediKev we cannot upgrade to 1.14.2 because we want to fix issue #5522 with timezone in release

**Comment by antiuser** [comment]
@JediKev yep, we are testing this moment and if agent is locked, notify about ticket agent receive on email anyway.
Ticket creating to a team (in this team we have 4 agents, and one agent is locked from their). But on this agent we have notification attribute

**Comment by JediKev** [comment]
@antiuser 

On Ticket create and on Assignment we loop through each staff and ensure they are available:
- https://github.com/osTicket/osTicket/blob/develop/include/class.ticket.php#L1762
- https://github.com/osTicket/osTicket/blob/develop/include/class.ticket.php#L2131

Is there anything special about this Agent? Is he Department Manager? Is he a Team Lead? Is his email the Admin Email? Any additional information will be helpful.

Cheers.

**Comment by antiuser** [comment]
@JediKev sorry for the late answer, yes this agent is Department Manager (not admin) and this agent is Team Lead too.
But we are tested with another usual agent (that locked and not included to Department Manager and Team Lead) and osTicket send email to this agent too
